projects/iot-cloud/software/lora2mqtt/src/main.py

```diff
@@ -0,0 +1,147 @@
+from flask import Flask
+from flask import jsonify
+from flask import request
+from flatten_json import flatten
+
+from utils.exceptions import RestException
+from utils.functions import resolve_mapping
+
+application = Flask(__name__)
+
+CONFIG_URL = os.environ.get('CONFIG_URL',
+                            'https://gist.githubusercontent.com/dorinelfilip/6743182de240959b2a723fcf232b8496/raw/'
+                            'f37c2be3e3e5f6abc54b16f14b7643ce0ab39d42/lora2mqtt.json')
+
+MQTT_HOST = os.environ.get('MQTT_HOST', 'mqtt.beia-telemetrie.ro')
+MQTT_PORT = os.environ.get('MQTT_PORT', 1883)
+
+
+@application.route('/')
+def hello_world():
+    return jsonify({'result': 'hello world'})
+
+
+@application.route('/publish', methods=['POST'])
+def publish_data():
+    data = request.get_json(silent=True)
+    if not data:
+        raise RestException(400, "Bad request", "Payload is not a valid JSON.")
+
+    try:
+        metrics = data['payload_fields']
+        metic_time = data['metadata']['time']
+        app_id = data['app_id']
+        dev_id = data['dev_id']
+    except KeyError:
+        raise RestException(400, 'Bad request', 'Mandatory key is missing from the JSON.')
+
+    authorization = request.headers.get('authorization')
+    if not authorization:
+        raise RestException(401, 'Not authorized', 'token authorization HTTP header is missing.')
+
+    mqtt_topic = resolve_mapping(application.config['mapping']['applications'], app_id, dev_id, authorization)
+
+    result = flatten(metrics)
+    result['timestamp'] = metic_time
+    payload = json.dumps(result)
+
+    application.logger.info('New topic: {}\nPayload: {}'.format(mqtt_topic, payload))
+
+    publish.single(mqtt_topic, hostname=MQTT_HOST, port=MQTT_PORT, payload=payload, qos=2)
+
+    return jsonify({'status': 'ok'})
+
+
+@application.before_first_request
+def init_app():
+    application.logger.setLevel(logging.INFO)
+    application.logger.info('Initializing application')
+    __print_environment_config()
+    application.logger.info('Downloading config')
+    downloaded_config = __get_config(silent=True)
+    if downloaded_config:
+        application.config['mapping'] = downloaded_config
+    else:
+        application.logger.error('Failed to download config.')
+        sys.exit(-1)
+
+    application.sched = scheduler = BackgroundScheduler(timezone='UTC')
+
+    @scheduler.scheduled_job('interval', seconds=300)
+    def refresh_config():
+        application.logger.info('Running periodic config refresh')
+        mew_config = __get_config(silent=True)
+        if not mew_config:
+            application.logger.warning('Failed to pull new config file')
+        else:
+            application.logger.info("New config loaded. Content {}".format(application.config['mapping']))
+            application.config['mapping'] = mew_config
+
+    scheduler.start()
+    atexit.register(lambda: clean_app(application))
+
+
+def __get_config(silent=False, timeout=5):
```

Comment:
Don't use `__` to prefix local functions.

Reply:
That is PEP8 for private/superprivate functions, not about exporting.

Reply:
It applies to classes and framework functions/methods. It just makes code harder to read.